TeamCode/src/main/java/org/firstinspires/ftc/teamcode/TeleOp.java

```diff
@@ -84,29 +79,25 @@ public void runOpMode() {
 
 //------------------Gripper Control-----------------
 
-            //Close the gripper
-            if (gamepad2.right_bumper && robot.gripperState != CLOSED && robot.gripperState != PWMDISABLED)  {
-                gripperPos = robot.gripperOneServo.getPosition() + GRIPPER_MODIFY;
-            }
-
-            //Open the gripper
-            if(gamepad2.right_trigger > 0.5 && robot.gripperState != OPEN && robot.gripperState != PWMDISABLED) {
-                gripperPos = robot.gripperOneServo.getPosition() - GRIPPER_MODIFY;
-            }
-
-            //automatically turn off the servo when grabbing a block
-            if(robot.gripperState == MIDDLE) {
-                robot.gripperOneServo.setPwmEnable();
+            if (robot.isGripperDisabled() == false) {
+                //Move the gripper
+                if (gamepad2.right_bumper)  {
+                    robot.moveGripperClosed();
+                } else if (gamepad2.right_trigger > 0.5) {
+                    robot.moveGripperOpen();
+                } else {
+                    robot.getGripperOneServo().setPwmDisable();
+                }
             } else {
-                robot.gripperOneServo.setPwmDisable();
+                robot.getGripperOneServo().setPwmDisable();
```

Comment:
I did dig into the code and found that setPwmDisable() was calling getPortNumber(), which returned portNumber. PortNumber had some code related to it. I wasn't sure if it would have a performance penalty, but just in case I made a variable for the port number and called controllerEx.setServoPwmDisable() from Robot.java. This also fixes the issue of accessing the servo object from other classes.